utils/lazy-pages-fuzzer/src/generate/mem_accesses.rs

```diff
@@ -0,0 +1,149 @@
+    fn default() -> Self {
+        InjectMemoryAccessesConfig {
+            max_accesses_per_func: 10,
+        }
+    }
+}
+
+#[derive(Debug, Display, Error, From)]
+pub enum InjectMemoryAccessesError {
+    #[display(fmt = "No memory imports found")]
+    NoMemoryImports,
+    #[display(fmt = "No code section found")]
+    NoCodeSection,
+    #[display(fmt = "")]
+    Arbitrary(arbitrary::Error),
+}
+
+// TODO: different word size accesses
+enum MemoryAccess {
+    ReadI32,
+    WriteI32,
+}
+
+pub struct InjectMemoryAccesses<'u> {
+    unstructured: Unstructured<'u>,
+    config: InjectMemoryAccessesConfig,
+}
+
+impl InjectMemoryAccesses<'_> {
+    pub fn new(
+        unstructured: Unstructured<'_>,
+        config: InjectMemoryAccessesConfig,
+    ) -> InjectMemoryAccesses<'_> {
+        InjectMemoryAccesses {
+            unstructured,
+            config,
+        }
+    }
+
+    fn generate_access_instructions(
+        u: &mut Unstructured,
+        target_addr: usize,
+    ) -> Result<Vec<Instruction>, InjectMemoryAccessesError> {
+        use MemoryAccess::*;
+        Ok(match u.choose(&[ReadI32, WriteI32])? {
+            ReadI32 => vec![
+                Instruction::I32Const(target_addr as i32),
+                Instruction::I32Load(0, 0),
+                Instruction::Drop,
+            ],
+            WriteI32 => vec![
+                Instruction::I32Const(target_addr as i32),
+                Instruction::I32Const(0xB6B6B6B6u32 as i32),
+                Instruction::I32Store(0, 0),
+            ],
+        })
+    }
+
+    pub fn inject<'this>(
+        mut self,
+        mut module: Module,
+    ) -> Result<(Module, Unstructured<'this>), InjectMemoryAccessesError>
+    where
+        Self: 'this,
+    {
+        let import_section = module
+            .import_section()
+            .ok_or(InjectMemoryAccessesError::NoMemoryImports)?;
+        let initial_memory_limit = import_section
+            .entries()
+            .iter()
+            .filter_map(|import| {
+                if let External::Memory(import) = import.external() {
+                    Some(import.limits().initial())
+                } else {
+                    None
+                }
+            })
+            .next()
+            .ok_or(InjectMemoryAccessesError::NoMemoryImports)?;
+
+        let code_section = module
+            .code_section_mut()
+            .ok_or(InjectMemoryAccessesError::NoCodeSection)?;
+
+        for function in code_section.bodies_mut() {
+            let access_count = self
+                .unstructured
+                .int_in_range(1..=self.config.max_accesses_per_func)?;
+
+            for _ in 0..=access_count {
+                let target_addr = self
+                    .unstructured
+                    .choose_index(initial_memory_limit as usize)
+                    .unwrap()
+                    .saturating_mul(OS_PAGE_SIZE);
+
+                let code_len = function.code().elements().len();
```

Comment:
same comment for the last position as in globals accesses injector

Reply:
It's correct, fn choose_index expect len of collection.